docs/4_dynamic.md

```diff
@@ -0,0 +1,166 @@
+    memorySizeGb: 1
+```
+
+> *Note: similar to a managed resource, there is no namespace defined on our
+> configuration for the `CloudMemorystoreInstanceClass` above because resource
+> classes are
+> [cluster-scoped](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition).*
+
+You will notice that this looks very similar to the
+`CloudMemorystoreInstanceClass` we created in the previous example. It has the
+same configuration for the external Cloud Memorystore instance, but you will
+notice that it contains a `specTemplate` instead of a `spec`. The `specTemplate`
+will be used later to create a `CloudMemorystoreInstance` that has the same
+configuration as the one in the static provisioning example.
+
+Create the `CloudMemorystoreInstanceClass`:
+
+```
+kubectl apply -f cloudmemorystoreclass.yaml
+```
+
+There is nothing to observe yet, we have published this configuration for a
+Cloud Memorystore instance for later use. To actually create our
+`CloudMemorystoreInstance`, we must create a `RedisCluster` claim that
+references the `CloudMemorystoreInstanceClass`. A claim can reference a class in
+two ways:
+
+1. **Direct Reference**: this is most similar to the way we referenced the
+   managed resource in the previous example. Instead of a `resourceRef` we can
+   provide a `classRef`:
+
+```yaml
+apiVersion: cache.crossplane.io/v1alpha1
+kind: RedisCluster
+metadata:
+  name: redis-claim-dynamic
+  namespace: crossplane-quickstart
+spec:
+  classRef:
+    apiVersion: cache.gcp.crossplane.io/v1beta1
+    kind: CloudMemorystoreInstanceClass
+    name: example-cloudmemorystore-class
+  writeConnectionSecretToRef:
+    name: redis-connection-details-dynamic
+```
+
+2. **Label Selector**: this is a more general way to reference a class, and also
+   requires less knowledge of the underlying implementation by the claim
+   creator. You will notice that the `CloudMemorystoreInstanceClass` we created
+   above includes the `guide: quickstart` label. If we include that label in a
+   selector on the `RedisCluster` claim, the claim will be scheduled to a class
+   that has that label.
+
+> *Note: if multiple classes have a label that is included in the claim's
+> selector, one will be chosen at random.*
+
+```yaml
+apiVersion: cache.crossplane.io/v1alpha1
+kind: RedisCluster
+metadata:
+  name: redis-claim-dynamic
+  namespace: crossplane-quickstart
+spec:
+  classSelector:
+    matchLabels:
+      guide: quickstart
+  writeConnectionSecretToRef:
+    name: redis-connection-details-dynamic
+```
+
+Using a label selector means that the `RedisCluster` claim creator is not
+concerned whether the claim is satisfied by a GCP `CloudMemorystoreInstance`, an
+AWS `ReplicationGroup`, an Azure `Redis`, or other. Each of these
+implementations will result in a Redis cluster being created and sufficient
+connection details being propagated to the namespace of the claim.
```

Comment:
It could be worth noting that the claim could be selecting between three different classes all within the same cloud - i.e. small, medium, or large CloudMemorystoreInstance not just AWS, GCP, or Azure Redis instance.

Reply:
Definitely, meant to work that in :)